app/controllers/training_and_support_controller.rb

```diff
@@ -10,34 +10,38 @@ def show
       raise Pundit::NotAuthorizedError, "This page is only available to CIP participants"
     end
 
+    current_user.participant_profile.update!(guidance_seen: true)
     @provider_name_key = @provider.name.downcase.gsub(/ /, "_")
   end
 
   def update
-    @participant = current_user.participant_profile
+    @guidance_speedbump_form = GuidanceSpeedbumpForm.new(guidance_question_params)
 
-    if params[@participant.model_name.param_key].blank?
-      @participant.errors.add :viewed_guidance, "Select if you would like a brief summary of the programme"
+    if @guidance_speedbump_form.view_guidance_option.blank?
+      @guidance_speedbump_form.errors.add :view_guidance_option, "Select if you would like a brief summary of the programme"
```

Comment:
This can be done by adding `validates :view_guidance_option, presence: { message: "Select if you would like a brief summary of the programme" }` in the form. And then here, you would say:

```
render :guidance_question and return unless @guidance_speedbump_form.valid?

# redirect code
```

Reply:
Thanks for this, i did try to validate it this way but was missing the valid method in the controller. I just assumed Rails would pick it up.